binary/reflect.go

```diff
@@ -803,6 +820,24 @@ func writeReflectJSON(rv reflect.Value, rt reflect.Type, w io.Writer, n *int64,
 			WriteTo([]byte("}"), w, n, err)
 		}
 
+	case reflect.Map:
+		WriteTo([]byte("{"), w, n, err)
+		keys := rv.MapKeys()
```

Comment:
If we were to support maps for the c-style-binary representation, then the nondeterminism would break the IAVLTree, should values support maps in structs, and the value codec were naive.